### ISSUE TITLE
Tasks does not support OC 9

### ISSUE THREAD

**Issue by JaBecker** [owncloud/tasks#296]
Im using OC 9.0.0.19 and Tasks 8.1.

When I open the tasks App theres just a blank page showing and the following errors appear in the owncloud error log:

`Error	PHP	Error: Class 'OC_Calendar_App' not found at /usr/share/webapps/owncloud/apps/tasks/controller/pagecontroller.php#60	2016-03-08T18:13:08+00:00
Error	PHP	Error: Class 'OC_Calendar_App' not found at /usr/share/webapps/owncloud/apps/tasks/controller/pagecontroller.php#60	2016-03-08T18:13:03+00:00
Error	PHP	Undefined index: newVersionString at /usr/share/webapps/owncloud/apps/updatenotification/templates/admin.php#8	2016-03-08T18:12:11+00:00
Error	PHP	Undefined index: newVersionString at /usr/share/webapps/owncloud/apps/updatenotification/templates/admin.php#8	2016-03-08T18:12:11+00:00` 

From what I've seen there are no errors in the php logfiles.

**Comment by Grot4x** [comment]
Same here OC: 9.0.0 Tasks: 0.8.1 Calendar: 1.0

**Comment by christian-putzke** [comment]
Same here :(

**Comment by raimund-schluessler** [comment]
Tasks 0.8.1 requires the old calendar app (https://github.com/owncloudarchive/calendar) which is now unmaintained and not supported for OC 9.

There currently is no Tasks version released which supports OC 9. However, I do work on a version, but it is not finished yet. You can track the state of the work in https://github.com/owncloud/tasks/pull/291. There you can see what is expected to work and what does not work yet. You can try the development version but be aware that it's not finished yet and may cause data loss in the worst case.

**Comment by JaBecker** [comment]
The Problem persists even when using version 0.9.0 from the DAVclient branch.

**Comment by raimund-schluessler** [comment]
I guess you did something wrong then. Please make sure, that you cleanly removed the old version of the tasks app. The version in the DAVclient branch does not call for ```OC_Calendar_App``` if it still does, you still have files of the old version in your apps folder.

**Comment by JaBecker** [comment]
I have removed the tasks app completely and installed the DAVclient branch again, still same error. Maybe the problem isn't actually related to the DAVclient but to something else that changed in OC9?

**Comment by raimund-schluessler** [comment]
Please check the error log again. I run the DAVclient branch successfully on OC 9.0.0RC3 without unexpected problems.

**Comment by JaBecker** [comment]
The following PHP error occurs now:

`TypeError: Argument 2 passed to OCA\Tasks\Controller\PageController::__construct() must be an instance of OCA\Tasks\Controller\IRequest, instance of OC\AppFramework\Http\Request given, called in /usr/share/webapps/owncloud/apps/tasks/appinfo/application.php on line 48 at /usr/share/webapps/owncloud/apps/tasks/controller/pagecontroller.php#37`

**Comment by raimund-schluessler** [comment]
Weird that this error did not show up on my test installation. However, I hope it is fixed now. Please checkout the current DAVclient (with commit https://github.com/owncloud/tasks/commit/2b1053e07203497991f24e238cc63e43dd71bf3e) and test it again.

**Comment by JaBecker** [comment]
Some new errors:

`Error: Call to undefined function OCA\Tasks\Controller\script() at /usr/share/webapps/owncloud/apps/tasks/controller/pagecontroller.php#62`

and

`Undefined variable: userSession at /usr/share/webapps/owncloud/apps/tasks/controller/pagecontroller.php#44	`

**Comment by raimund-schluessler** [comment]
I fixed the second error and some other things I just found. I will need to take a closer look at the first error (already reported in https://github.com/owncloud/tasks/issues/295)

**Comment by raimund-schluessler** [comment]
Please try again with current DAVclient version https://github.com/owncloud/tasks/commit/c8888cbfac0662d51223a632c0bd2cfbeee499d4

**Comment by JB1985** [comment]
@raimund-schluessler your changes not work for me.

Error: 
PHP Fatal error:  Call to a member function getAppValue() on null in /var/customers/webs/web12/owncloud/apps/tasks/controller/pagecontroller.php on line 56

**Comment by raimund-schluessler** [comment]
Which OC version do you use?

**Comment by JB1985** [comment]
OC9.0

**Comment by JaBecker** [comment]
Now its marginally better: the top ribbon - where the owncloud logo is - is shown now. New errors, this time "tasks" errors:

`Could not find resource file "/owncloud/apps/tasks/js/vendor/ical.js/build/ical.js"`

`Could not find resource file "/owncloud/apps/tasks/js/vendor/angular-ui-select/dist/select.min.js"`

**Comment by raimund-schluessler** [comment]
@JaBecker You are right, these files did not exist in the repository since the .gitignore prevented commiting them. Only a ```bower install``` would have loaded them from their respecitve source repositories. I fixed it and they are available in the DAVclient branch now. Would be cool if you could test again.

@JB1985 I am also on OC v9.0.0 and the DAVclient branch works without debug or error messages. Can you please check, that all permissions are set correctly?

**Comment by tchncs** [comment]
Wonderful, switched to the DAVclient branch all also issues from before are fixed :) <3
-> #285

**Comment by JB1985** [comment]
@raimund-schluessler the permissions are correct.

I have install the Tasks from OC with the Version 0.8.1.

I have try again. Delete tasks folder and install it again over OC. Overwrite the two files 

https://raw.githubusercontent.com/owncloud/tasks/c8888cbfac0662d51223a632c0bd2cfbeee499d4/controller/pagecontroller.php
https://raw.githubusercontent.com/owncloud/tasks/c8888cbfac0662d51223a632c0bd2cfbeee499d4/templates/main.php

and get the server error 500 and in the logs are: 

PHP Fatal error:  Call to a member function getAppValue() on null in /var/customers/webs/web12/owncloud/apps/tasks/controller/pagecontroller.php on line 56

Line 56: 

 $appVersion = $this->config->getAppValue($this->appName, 'installed_version');

**Comment by JaBecker** [comment]
Now everything works as advertised! Good Work!

**Comment by raimund-schluessler** [comment]
@tchncs @JaBecker Good to hear :smile: 

@JB1985 Tasks 0.8.1. does not work on OC 9.0.0. Overwriting some files won't fix this. You have to use *all* files from the DAVclient branch https://github.com/owncloud/tasks/tree/DAVclient as discussed above.

**Comment by JB1985** [comment]
@raimund-schluessler 

all right, now its working! Why in OC Apps is a old App?

**Comment by raimund-schluessler** [comment]
@JB1985 There currently is no release of Tasks for oc 9. There will be one once the DAVclient branch is merged. The old version in OC apps would still work on oc 9 if you installed the old calendar app, but they don't use the built-in dav server of oc 9 and it is not advised to use them. I will adjust the max-version of the old app to prevent installing them on oc 9 from the appstore.

**Comment by binarykitchen** [comment]
ok, what is the easiest solution here to fix this for v9 with few commands? have installed via apt-get ...

**Comment by JB1985** [comment]
cd /var/www/owncloud/apps
rm tasks
wget https://github.com/owncloud/tasks/archive/DAVclient.zip
mv ....master tasks

Check the permissions and be happy :+1:

**Comment by kenernest** [comment]
Hi, sorry for my english. it isn`t my native languaje.

I have a trouble with the solution. When I activated task 0.90 the link into owncloud menu has dissapeared and deactivated again. I look into registry log and i had  "No update found at the ownCloud appstore for app 166049"

Im solved with a little mix and now already is ok. I have created a folder named tasks with all task 0.8.1 files. After I unzipped the file https://github.com/owncloud/tasks/archive/DAVclient.zip and rewrite the archives. 

Now tasks is on fire.

**Comment by marvin21** [comment]
@kenernest did you move all files from tasks-DAVclient into the folder tasks?

**Comment by binarykitchen** [comment]
oh i see thanks.

```
cd /var/www/owncloud/apps
rm -rf tasks
git clone https://github.com/owncloud/tasks tasks
```

this worked well for me

**Comment by binarykitchen** [comment]
but now i see a new error :(

```
{"reqId":"EdR7ApC8a+oRCOGguohu","remoteAddr":"203.84.135.120","app":"PHP","message":"Class 'OC_Calendar_App' not found at \/var\/www\/owncloud\/apps\/tasks\/controller\/pagecontroller.php#60","level":3,"time":"2016-03-10T21:53:30+00:00"}
```

**Comment by binarykitchen** [comment]
hmm, nothing works :(

**Comment by raimund-schluessler** [comment]
@binarykitchen You still have the old Tasks app installed. The DAVclient branch does not ask for ```OC_Calendar_App```. Make sure, that all files are replaced properly and the old version is removed completely.

**Comment by binarykitchen** [comment]
@raimund-schluessler oh right, fixed that. works now

**Comment by dper** [comment]
@raimund-schluessler I just installed from the zip file as you described above. Works great. Thanks! :-)

**Comment by devuan2** [comment]
worked for me (tasks app was  already disabled/uninstalled in owncloud gui)
```
sudo -u www-data bash
cd /var/www/owncloud/apps
git clone https://github.com/owncloud/tasks tasks
git checkout DAVclient

```
Tasks up and running in owncloud 9.0

**Comment by mhzawadi** [comment]
Hi all,

I have OC9 in testing and had found this app not working, but have just downloaded the update and that does work for the most part.

But the comment thing doesn't work, I get an error in the OC log

{"reqId":"Hjs4z14CZQeKo3asK6AK","remoteAddr":"192.168.49.6","app":"PHP","message":"Automatically populating $HTTP_RAW_POST_DATA is deprecated and will be removed in a future version. To avoid this warning set 'always_populate_raw_post_data' to '-1' in php.ini and use the php:\/\/input stream instead. at Unknown#0","level":3,"time":"2016-03-12T15:54:35+00:00","method":"POST","url":"\/index.php\/apps\/tasks\/tasks\/20150612T202310Z-27668%407fbc1cf7e9dc1edb.ics\/comment"}

**Comment by raimund-schluessler** [comment]
As discussed above, the DAVclient branch is *not* finished yet. You can see that works at https://github.com/owncloud/tasks/pull/291

**Comment by zenlord** [comment]
For the sake of completeness: @devuan2 had it almost correct. I had to perform one intermediate step:
```
sudo -u www-data bash
cd /var/www/owncloud/apps
rm -rf tasks/
git clone https://github.com/owncloud/tasks tasks
cd tasks
git checkout DAVclient
```
After which you can enable the tasks-app in the ownCloud interface, and it will immediately take you to an update page to update the tasks-app.

**Comment by zaphbbrox** [comment]
```
cd /var/www/owncloud/apps
rm -rf tasks
wget https://github.com/owncloud/tasks/archive/DAVclient.zip
```

Thx this works great, but now the app is a little bit slower than before.

**Comment by raimund-schluessler** [comment]
@zaphbbrox See my comment https://github.com/owncloud/tasks/issues/302#issuecomment-199179350

**Comment by kwoxer** [comment]
When will the Tasks app be officially rereleased and fixing that issue. Is there already something concrete about that?

**Comment by spoorun** [comment]
My trust is happy to pay on a bounty on helping get fixed and release for 9.0. :)

**Comment by kwoxer** [comment]
Still no official fix for Tasks? This is so cool and important guys. Really why isn't this fixed ASAP? I know there is also some other stuff, but there is no alternative to this and it worked great in 8.2. Please let me know about any progress :)

**Comment by triplesek** [comment]
@kwoxer Check out the `DAVclient` branch, works for me without issues or workarounds.

**Comment by raimund-schluessler** [comment]
@olantrust Glad to see that there are people caring for this app. You can place a bounty and see if more developers are willing to work on this. However, for me it is about time, not about money (although I wouldn't say "No" to a bounty :wink:)

@kwoxer This is a hobby project which is currently only maintained by one developer (me :smirk:). I can only work on this in my freetime. Porting to OC9 is a huge project as the app has to be nearly completly rewritten. You will have to use the DAVclient branch and do some hand work for now.

I will release a version of the app for OC9 once I manage to implement coping with a lot of tasks. Currently the app is slow, if there are very many tasks and I consider this a showstopper. The first version for OC9 will lack support of reminders, comments and won't have a search function. This will be reimplemented later on, but I guess it is better than not having a version at all.

**Comment by ENDev15** [comment]
Thanks, worked great with the DAVClient branch.

**Comment by Apkahym** [comment]
For my work this:

```
rm -R tasks/
git clone https://github.com/owncloud/tasks tasks
cd tasks
git checkout DAVclient
```
Thanks

**Comment by boospy** [comment]
Thanks a lot for the Update, works here fine on Owncloud 9.x

```
sudo -u www-data bash
cd /var/www/owncloud/apps
rm -rf tasks/
git clone https://github.com/owncloud/tasks tasks
cd tasks
git checkout DAVclient
```

**Comment by kwoxer** [comment]
Jap, @boospy 's and @Apkahym works. Tasks again working in my browser. Thank you guys.

Just got `➜  tasks git:(master) git checkout DAVclient
Branch DAVclient set up to track remote branch DAVclient from origin.
Switched to a new branch 'DAVclient'`

But I think this is fine right?

**Comment by valentin8709** [comment]
Everything is working fine, thank you very much!! Nevertheless, I just have an "integrity checking" problem. Do you know how to resolve this? Thanks again =)

**Comment by strugee** [comment]
@valentin8709 you need to provide way more information for anyone to have a chance of helping you. Like, you know, the actual error message you got.

**Comment by valentin8709** [comment]
@strugee I just installed the app as printed:
sudo -u www-data bash
cd /var/www/owncloud/apps
rm -rf tasks/
git clone https://github.com/owncloud/tasks tasks
cd tasks
git checkout DAVclient

And I have an error while checking the core integrity (the new feature in oc 9). Here is the log:

Technical information
=====================
The following list covers which files have failed the integrity check. Please read
the previous linked documentation to learn more about the errors and how to fix
them.

Results
=======
- core
	- INVALID_HASH
		- .htaccess
		- .user.ini
		- lib/private/group/manager.php
	- EXTRA_FILE
		- updater/update.log
- external
	- EXTRA_FILE
		- img/ampache.ico
		- img/lychee.ico
- activity
	- INVALID_HASH
		- lib/mailqueuehandler.php

Raw output
==========
Array
(
    [core] => Array
        (
            [INVALID_HASH] => Array
                (
                    [.htaccess] => Array
                        (
                            [expected] => f40e7ca19cd76ff922ccd164e38f4d73806f66170cafa202fb470be225c813b4b242f4321da8a01bef1329e300acf3d137dbcbff058c501f166a2619031fc651
                            [current] => 5e26947c6350e499cad559198879ee054acb0bb1dc244d02655c9f6d824276ac52c15ff913490ed670b1216ce012b3bdf8307a88b16e0da009fa0641261378d9
                        )

                    [.user.ini] => Array
                        (
                            [expected] => 0a557e3cdca4c2e3675deed761d79d109011dcdebbd9c7f6429f1d3476938ec95729543d7384651d1d0c48e26c5024cc5f517445920915a704ea748bdb903c5f
                            [current] => f5f88f2834b032e504c3447bcd7aafb4611895d4e43cb5795e56696c1e525e68535888ded947322f5520ae973484aa740f27f173a7a260176f62dd7d59e6ac90
                        )

                    [lib/private/group/manager.php] => Array
                        (
                            [expected] => 6ea2a0e30aa27f459d65749ca1513f7745b18a2d164066aa377d184550311e3f7802092b81cba192b2d9a6e711a7cb102470e5947556ad6890b892caa13e12eb
                            [current] => d945197ea45090da81d53704449452a1ba2c5fc94a2f26cae7d9623d519d70d38546062e88cf4b4a2a780035bb97e1ef189fd21810e8b7069ecdfb5569f0b894
                        )

                )

            [EXTRA_FILE] => Array
                (
                    [updater/update.log] => Array
                        (
                            [expected] => 
                            [current] => 62f7024a4836f36e49b671fef20ce5837f0682ff9b7d0d09147797e751c6092de1c51ad517cb05ba0d7a3661414a6f82f572bf1207f7ec3deca847ccb798697b
                        )

                )

        )

    [external] => Array
        (
            [EXTRA_FILE] => Array
                (
                    [img/ampache.ico] => Array
                        (
                            [expected] => 
                            [current] => 5e5064d4a39762b58249d4d431227ef21beaa0953e05662b8fc093990f1f936d33083c67489af32b85fa95b4ded66daa77b482ae013c5401ae9c32c6066b2d45
                        )

                    [img/lychee.ico] => Array
                        (
                            [expected] => 
                            [current] => 4bca3c42f1f3cfcac2e9c7b243bad0ef6fba6a446b8d7f50f7bb832615d4e9730782ae20e09ef18d61c4fee00a33ee0deabd6d249684d8969921891451d59f03
                        )

                )

        )

    [activity] => Array
        (
            [INVALID_HASH] => Array
                (
                    [lib/mailqueuehandler.php] => Array
                        (
                            [expected] => 44c62d3d282ada7263b3c8883be36133e1decd080455ceaf85cb514669a6aaf031b8449460389effe8a079edd9c5b37514308a92a5aac4f4638e52a41bb03c08
                            [current] => 581a57bbb41b4834592a74a0c3502303a0c25af5a533a33a3d9edf6c4dd01bd165daf3d229ff8d81ec2fbe6a4c6da399f3ffe4a7ef7f50948ea30ddd1da9f5b7
                        )

                )

        )

)

**Comment by strugee** [comment]
@valentin8709 that's much better.

Those errors don't look like they're related to Tasks to me. They're talking about completely separate apps.

**Comment by valentin8709** [comment]
Impossible, I have this error since I activated the app (and haven't when I disable it). I will looking for a solution when I'll have time

**Comment by e-alfred** [comment]
Any chances that the new Tasks app for OC9 will support IE11/Edge?

**Comment by valo-lindstroem** [comment]
I can confirm that the DAVclient branch works on OC 9.0.2.

I had to enable the app with the 'occ app:enable tasks' command, it was not listet in the not enabled apps after i switched to the DAVclient branch and reloaded OC.

**Comment by raimund-schluessler** [comment]
To install DAVclient branch manually via bash:

```bash
cd /var/www/owncloud/apps
rm -rf tasks
wget https://github.com/owncloud/tasks/archive/DAVclient.zip
unzip DAVclient.zip
mv tasks-DAVclient tasks
rm DAVclient.zip
sudo -u www-data php occ app:enable tasks
sudo -u www-data php occ upgrade
```

**Comment by olaulau** [comment]
I'm affected too.
DAVclient.zip works but gives a warning about code integrity check when logged with admin user.
and I wonder how will be futures updates with this app installed ...

**Comment by raimund-schluessler** [comment]
I merged #291 into master just yet and will prepare a release for the appstore soon.

@olaulau The official way is to install via the appstore which will be possible once I release the version to the appstore. Also future updates will be released this way and the code integrity check should work then too.

To switch to the appstore version you have to deactive and remove the DAVclient version and then just normally install via the appstore.

**Comment by raimund-schluessler** [comment]
Can be released once https://github.com/owncloud/appstore-issues/issues/62 is done. By then: please test Tasks master!

**Comment by caswi** [comment]
Old Tasks-App deactivated and removed.
Updated Tasks-App via FTP uploaded and activated => App is working fine
Thanks a lot for great job!

**Comment by raimund-schluessler** [comment]
I just released Tasks v0.9. You can install the app via Appstore again.

**Comment by Paddo5819** [comment]
Any suggestion when the commemt feature is implemented? That would be for our work the most important feature. I installed the 0,9 version atm and we are very impressed ☺

**Comment by kwoxer** [comment]
Gonna test it this evening. Thank you. :)

**Comment by gw0** [comment]
Thank you, seems to work. :)

**Comment by mhzawadi** [comment]
looks to be work now, good job :)

**Comment by kwoxer** [comment]
Mhh well I unstalled that old version that I got from Github.

But I cannot find any App called **Tasks** in the ownCloud Backend. Maybe it's still cached or so? For sure I deleted the **/tasks** folder.

**Comment by dper** [comment]
@kwoxer If you're asking where to download the app... https://apps.owncloud.com/content/show.php/Tasks?content=164356

**Comment by kwoxer** [comment]
So it is not anymore available in my ownCloud backend as before, is that correct?

**Comment by gw0** [comment]
I did not need to do anything manually. Under Apps there was an Upgrade button for Tasks app and it worked flawlessly.

**Comment by vinilox** [comment]
If it can be of any help, here is how I did "git version → app store version" : 

1. Disable Tasks app on owncloud webUI
2. rm -rf apps/tasks
3. Install Tasks app with owncloud webUI

**Comment by raimund-schluessler** [comment]
The Tasks app should appear in the Productivity section of your owncloud apps without you doing anything. If it for sure does not, then there might be something wrong with your owncloud installation.

**Comment by kwoxer** [comment]
Strange, just after upgrade from 9.0.1 to 9.0.2 the Tasks app was available in the owncloud backend again. Now it's working properly. Thank you.
